### ISSUE TITLE
Units error in icrs_to_helioecliptic (in icrs_to_helioecliptic)

### ISSUE THREAD

**Issue by dcunning11235** [astropy/astropy#4358]
In [line 76 of ecliptic_transfprms.py](https://github.com/astropy/astropy/blob/v1.1.x/astropy/coordinates/builtin_frames/ecliptic_transforms.py#L76) there appears to be a bug:

units.au is multiplied with a result and then added to a scalar; I believe there are simply missing parentheses here.  This then creates/reveals a dimensions issue (at least when used with via calling `transform_to` on an array.)   What worked for me was adding two changes.  Replacing the current line 76 with:

`heliocart = ((from_coo.cartesian.xyz).T + delta_bary_to_helio) * u.au`

and then updating line 82 to be:

`newrepr = cartrepr_from_matmul(rmat, ICRS(CartesianRepresentation(heliocart.T)))`

I did some spot checking of this on a transformed array of coordinates and the handful I checked appear correct.

**Comment by eteq** [comment]
@dcunning11235 - can you give a snippet of code that actually triggers this error? If you look at [test_ecliptic.py](https://github.com/astropy/astropy/blob/v1.1.x/astropy/coordinates/tests/accuracy/test_ecliptic.py#L37) you'll see that this actually *is* run by the tests, and seems to work.

Is it possible you're trying to transform a SkyCoord that does not have a distance?  I.e., not the following:
```
>>> scwd = SkyCoord(1*u.deg,2*u.deg,distance=1.5*u.au)
>>> scwd.heliocentrictrueecliptic
<SkyCoord (HeliocentricTrueEcliptic: equinox=J2000.000): (lon, lat, distance) in (deg, deg, AU)
    (1.80738559, 1.42245507, 1.50721166)>
>>> sc = SkyCoord(1*u.deg,2*u.deg)
>>> sc.heliocentrictrueecliptic
UnitConversionError: Can only apply 'add' function to quantities with compatible dimensions
```
This is *expected* behavior: ``from_coo.cartesian.xyz`` normally yields something with length dimensions, but if there's no distance, then it comes out instead as dimensionless, which then can't combine with something that has AU dimensions.  But if you use a coordinate that has a distance (i.e., a point in 3D space instead of on the unit sphere), it should work.

The dimensionality problem looks to be real, though.  So I think your fix will help if you remove the outer parenthesis.  @dcunning11235, do you want to try to create a pull request to address this?  The advantage to that (instead of me just fixing it myself) is that you get added to the list of astropy contributors.  (But note that this is pretty urgent because we want to get v1.1 released ASAP...)

**Comment by dcunning11235** [comment]
You are correct @eteq, I was using SkyCoord's created with only RA and Dec.

I'd be be happy to create a pull request, just including the `.T` changes.  Will do it before the end of my day.

**Comment by eteq** [comment]
@dcunning11235 - OK, great.  If you feel up to it you might also add a regression test to test the case that fails (which I think is for array coordinates w/distances, right?) - but if you're not sure how/don't want to do that, I can do it when I merge the fix.

Thanks!

**Comment by dcunning11235** [comment]
Sure, @eteq, not a big addition (to an already small task :>)  But I'll say "not until tomorrow" since I haven't looked at the Astropy unit tests at all, yet.

**Comment by eteq** [comment]
@dcunning11235 - note that in #4371 I added to your change in #4367.  That's been merged now, so this got closed, but if you're still seeing related issues after that, feel free to reopen this (or create a new issue).
